base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -1582,12 +1582,7 @@ fn rewind_to_height<T: BlockchainBackend>(db: &mut T, height: u64) -> Result<Vec
     // Delete blocks
     let metadata = db.fetch_chain_metadata()?;
     let block_height = metadata.height_of_longest_chain();
-    let steps_back = block_height.checked_sub(height).ok_or_else(|| {
-        ChainStorageError::InvalidQuery(format!(
-            "Cannot rewind to height ({}) that is greater than the block height {}.",
-            height, block_height
-        ))
-    })?;
+    let steps_back = block_height.saturating_sub(height);
```

Comment:
I think it needs to return here, otherwise it will set the chain metadata to the tip at line 1608, which may be higher than the current tip

Reply:
Ah you're right - fixed